staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -291,6 +292,43 @@ func (*Mounter) List() ([]MountPoint, error) {
 	return ListProcMounts(procMountsPath)
 }
 
+const (
+	// statTimeout avoids the risk
+	// of os.Stat getting hung forever
+	// when called on something such as
+	// an unresponsive NFS volume
+	statTimeout = time.Minute
+)
+
+// statWithTimeout is a wrapper around os.Stat which returns an error if
+// os.Stat takes more time to finish execution than the specified timeout.
+func statWithTimeout(file string, timeout time.Duration) (fs.FileInfo, error) {
+	// if this channel receives, then the command completed and did not
+	// hang beyond the set timeout, completion could've been successful
+	// or unsuccessful
+	done := make(chan error)
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+
+	var info fs.FileInfo
+	var err error
+	go func() {
+		info, err = os.Stat(file)
```

Comment:
I think this will leave the go routines hanging around on a nfs `hard` mount when the server is gone.
So overtime you will end up having a bunch of them hanging around.

One can replace the direct os.stat call with an invocation of the stat process. Since then the process can be killed.
I don't know of a way to forcefully terminate a go routine? See intr comment for newer kernels.
https://linux.die.net/man/5/nfs

I use the stat process approach for our CSI driver when dealing with NFS mounts.
It might be to costly for running it in a reconcile loop though, would need some testing I guess.
https://github.com/longhorn/longhorn-manager/blob/51c3b60138b0ca85c8bbfdffa4cc6998844246a8/csi/nfs/nsenter_mount.go#L270

One can also use the /proc/mounts checks like the host util does.
https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/hostutil/hostutil_linux.go#L162
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/mount-utils/mount_helper_unix.go#L87

This isn't great either since reading from /proc/mounts has some caveats mentioned here:
https://github.com/kubernetes/kubernetes/issues/31272#issuecomment-300209720

Reply:
This reminds me, we *may* have https://github.com/kubernetes/kubernetes/issues/102664 bug

Reply:
@MadhavJivrajani would you try @joshimoo 's approach and update the PR?

Reply:
Hey, yes I will update the PR over the weekend. Thanks!

Reply:
@MadhavJivrajani Are you planning to work on updating this PR btw?

Reply:
Hey, I'm sorry - I'm not able to find the time to work on this. Please feel free to close this PR if needed. 

I apologise again for the delay.